reference/test/factorization/ilu_kernels.cpp

```diff
@@ -0,0 +1,505 @@
+    using Dense = typename TestFixture::Dense;
+    const auto x = gko::initialize<Dense>({1., 2., 3.}, this->exec);
+    auto b_lu = Dense::create_with_config_of(gko::lend(x));
+    auto b_ref = Dense::create_with_config_of(gko::lend(x));
+
+    auto factors = this->ilu_factory_skip->generate(this->mtx_small);
+    factors->apply(gko::lend(x), gko::lend(b_lu));
+    this->mtx_small->apply(gko::lend(x), gko::lend(b_ref));
+
+    GKO_ASSERT_MTX_NEAR(b_lu, b_ref, r<value_type>::value);
+}
+
+
+TYPED_TEST(Ilu, GenerateForDenseSmall)
+{
+    using value_type = typename TestFixture::value_type;
+    auto factors = this->ilu_factory_skip->generate(this->mtx_small);
+    auto l_factor = factors->get_l_factor();
+    auto u_factor = factors->get_u_factor();
+
+    GKO_ASSERT_MTX_NEAR(l_factor, this->small_l_expected, r<value_type>::value);
+    GKO_ASSERT_MTX_NEAR(u_factor, this->small_u_expected, r<value_type>::value);
+}
+
+
+TYPED_TEST(Ilu, GenerateForCsrSmall)
+{
+    using value_type = typename TestFixture::value_type;
+    auto factors = this->ilu_factory_skip->generate(this->mtx_csr_small);
+    auto l_factor = factors->get_l_factor();
+    auto u_factor = factors->get_u_factor();
+
+    GKO_ASSERT_MTX_NEAR(l_factor, this->small_l_expected, r<value_type>::value);
+    GKO_ASSERT_MTX_NEAR(u_factor, this->small_u_expected, r<value_type>::value);
+}
+
+
+TYPED_TEST(Ilu, GenerateForCsrSmall2ZeroDiagonal)
+{
+    using value_type = typename TestFixture::value_type;
+    auto factors = this->ilu_factory_skip->generate(this->mtx_csr_small2);
+    auto l_factor = factors->get_l_factor();
+    auto u_factor = factors->get_u_factor();
+
+    GKO_ASSERT_MTX_NEAR(l_factor, this->small2_l_expected,
+                        r<value_type>::value);
+    GKO_ASSERT_MTX_NEAR(u_factor, this->small2_u_expected,
+                        r<value_type>::value);
+}
+
+
+TYPED_TEST(Ilu, GenerateForCsrBigWithDiagonalZeros)
+{
+    using value_type = typename TestFixture::value_type;
+    auto factors = this->ilu_factory_skip->generate(this->mtx_big_nodiag);
+    auto l_factor = factors->get_l_factor();
+    auto u_factor = factors->get_u_factor();
+
+    GKO_ASSERT_MTX_NEAR(l_factor, this->big_nodiag_l_expected,
+                        r<value_type>::value);
+    GKO_ASSERT_MTX_NEAR(u_factor, this->big_nodiag_u_expected,
+                        r<value_type>::value);
+}
+
+
+TYPED_TEST(Ilu, GenerateForDenseBig)
+{
+    using value_type = typename TestFixture::value_type;
+    auto factors = this->ilu_factory_skip->generate(this->mtx_big);
+    auto l_factor = factors->get_l_factor();
+    auto u_factor = factors->get_u_factor();
+
+    GKO_ASSERT_MTX_NEAR(l_factor, this->big_l_expected, r<value_type>::value);
+    GKO_ASSERT_MTX_NEAR(u_factor, this->big_u_expected, r<value_type>::value);
+}
+
+
+TYPED_TEST(Ilu, GenerateForDenseBigSort)
+{
+    using value_type = typename TestFixture::value_type;
+    auto factors = this->ilu_factory_skip->generate(this->mtx_big);
```

Comment:
```suggestion
    auto factors = this->ilu_factory_sort->generate(this->mtx_big);
```

Reply:
Great catch, that was also an issue in the ParILU tests